aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -279,9 +288,24 @@ public void onResult(SignUpResult result) {
 
                 @Override
                 public void onError(Exception error) {
```

Comment:
I think it will be useful to create an `AuthExceptionUtil` class or something along that line to help declutter the plugin code.

See [how iOS handles them](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Support/Utils/AuthErrorHelper.swift) for example.

Although, it will end up looking quite different compared to iOS. I imagine that the class will end up looking like:
```
public final class AuthExceptionUtil {
    private AuthExceptionUtil() {}
    public static AuthException resolveSignInError() {
        //...your code here
    }
    public static AuthException resolveSignOutError() {
        //...your code here
    }
    //...more methods to resolve each auth operation
}
```